flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -180,10 +174,10 @@ public InputStatus processInput() throws Exception {
 
 		if (readingInputIndex == 0) {
 			firstInputStatus = processor1.processInput();
-			checkFinished(firstInputStatus);
+			inputSelectionHandler.nextSelection();
 		} else {
 			secondInputStatus = processor2.processInput();
-			checkFinished(secondInputStatus);
+			inputSelectionHandler.nextSelection();
```

Comment:
nitty nit: move `inputSelectionHandler.nextSelection();` outside of the if/else?